app/main.py

```diff
@@ -1,13 +1,41 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "column": error["column_number"], "line": error["line_number"],
+        "message": error["text"], "name": error["code"], "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [
+            {
+                "line": error["line_number"],
+                "column": error["column_number"],
+                "message": error["text"],
+                "name": error["code"],
+                "source": "flake8"
+            } for error in errors
+        ],
+        "path": file_path, "status": "failed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "errors": [] if len(linter_report[reports]) == 0 else [
+                {
+                    "line": error["line_number"],
+                    "column": error["column_number"],
+                    "message": error["text"],
+                    "name": error["code"],
+                    "source": "flake8"
+                }
+                for error in linter_report[reports]
+            ],
+            "path": reports,
+            "status": "passed" if len(linter_report[reports]) == 0
+            else "failed"
+        }
+        for reports in linter_report
```

Comment:
A few notes:
1. When you loop through dict, you get keys without their values. You can use dict.items() and it returns (key, value). In this way you can loop like:
for key, value in dict.items(): ...
2. Using #1 you can simplify linter_report[reports] statements
3. Using #1 you have values and can simplify if statement that you use for "status" key
4. Also note that this formatting is already implemented in your functions, so you can re-use it

Reply:
I understand and have refactor my code)))

I agree with all improvements, don`t know why I don`t seen it before myself, yet.)